File: client_servo_button.py
#!/usr/bin/python3

import socket
import time
import sys
from PyQt5.QtWidgets import (QWidget, QMainWindow, QGridLayout, QPushButton, QApplication)

HOST = '192.168.86.207'   # The remote host
PORT = 50007              # The same port as used by the server

sock=socket.socket(socket.AF_INET, socket.SOCK_STREAM)
sock.connect((HOST, PORT))

class Client_servo_button(QMainWindow):
    
    def __init__(self):
        super().__init__()
        
        self.initUI()
        
        
    def initUI(self):      

        self.widget = QWidget()
        self.setCentralWidget(self.widget)

        leftbtn = QPushButton("Left", self)
        middlebtn = QPushButton("Middle", self)
        rightbtn = QPushButton("Right", self)
        stopbtn = QPushButton("Stop", self)

        grid = QGridLayout(self.widget)
        grid.setSpacing(10)
        grid.addWidget(leftbtn, 0, 0)
        grid.addWidget(middlebtn, 0, 1)
        grid.addWidget(rightbtn, 0, 2)
        grid.addWidget(stopbtn, 1, 0)

        self.widget.setLayout(grid)
      
        leftbtn.clicked.connect(self.buttonClicked)            
        middlebtn.clicked.connect(self.buttonClicked)
        rightbtn.clicked.connect(self.buttonClicked)
        stopbtn.clicked.connect(self.buttonClicked)
        
        self.statusBar()
        
        self.setGeometry(300, 300, 290, 200)
        self.setWindowTitle('client servo button')
        self.show()
        
        
    def buttonClicked(self):
      
        sender = self.sender()
        if sender.text() == "Left":
            sock.sendall(b"left")
            data = sock.recv(1024)
            self.statusBar().showMessage(repr(data))
        elif sender.text() == "Middle":
            sock.sendall(b"middle")
            data = sock.recv(1024)
            self.statusBar().showMessage(repr(data))
        elif sender.text() == "Right":
            sock.sendall(b"right")
            data = sock.recv(1024)
            self.statusBar().showMessage(repr(data))
        elif sender.text() == "Stop":
            sock.sendall(b"stop")
            data = sock.recv(1024)
            self.statusBar().showMessage(repr(data))
        
if __name__ == '__main__':
    
    app = QApplication(sys.argv)
    ex = Client_servo_button()
    sys.exit(app.exec_())

File: client_servo_button.cpp
#include "client_servo_button.h"
#include <QGridLayout>
#include <QStatusBar>
#include <stdio.h>
#include <sys/socket.h>
#include <arpa/inet.h>
#include <netinet/in.h>
#include <netdb.h>
#include <unistd.h>
#include <time.h>
#include <string.h>
#include <stdlib.h>

#define HOST "192.168.86.207"
#define PORT 8888

int make_socket()
{
    int sock;
    struct sockaddr_in serv_addr;

    if ((sock = socket(AF_INET, SOCK_STREAM, 0)) < 0)
    {
        printf("\n Socket creation error \n");
        return -1;
    }

    memset(&serv_addr, '0', sizeof(serv_addr));

    serv_addr.sin_family = AF_INET;
    serv_addr.sin_port = htons( PORT );

    // Convert IPv4 and IPv6 addresses from text to binary form
    if(inet_pton(AF_INET, HOST, &serv_addr.sin_addr)<=0)
    {
        printf("\nInvalid address/ Address not supported \n");
        return -1;
    }

    if (connect(sock, (struct sockaddr *)&serv_addr, sizeof(serv_addr)) < 0)
    {
        printf("\nConnection Failed \n");
        return -1;
    }
    return sock;
}
   
Client_servo_button::Client_servo_button(QWidget *parent)
    : QMainWindow(parent) {

  QWidget* widget = new QWidget(this);

  QPushButton *servoRightBtn = new QPushButton("Right", this);
  QPushButton *servoLeftBtn = new QPushButton("Left", this);
  QPushButton *servoMiddleBtn = new QPushButton("Middle", this);
  QPushButton *servoStopBtn = new QPushButton("Stop", this);
  
  QGridLayout *grid = new QGridLayout(widget);
  grid->addWidget(servoRightBtn, 0, 2);
  grid->addWidget(servoLeftBtn, 0, 0);
  grid->addWidget(servoMiddleBtn, 0, 1);
  grid->addWidget(servoStopBtn, 1, 0);

  widget->setLayout(grid);
  setCentralWidget(widget);
  statusBar();

  connect(servoRightBtn, &QPushButton::clicked, this, &Client_servo_button::servoRight);
  connect(servoLeftBtn, &QPushButton::clicked, this, &Client_servo_button::servoLeft);
  connect(servoMiddleBtn, &QPushButton::clicked, this, &Client_servo_button::servoMiddle);
  connect(servoStopBtn, &QPushButton::clicked, this, &Client_servo_button::servoStop);
  sock = make_socket();
}

void Client_servo_button::send_servo( const char * servo ) {

  int valread;
  char buffer[1024] = {0};
  send(sock , servo, strlen(servo) , 0 );
  valread = read( sock, buffer, 1024 );
  buffer[valread] = '\0';
  statusBar()->showMessage(buffer);
}

void Client_servo_button::servoRight() {
  const char *servo = "right";
  send_servo( servo );
}

void Client_servo_button::servoLeft() {
  const char *servo = "left";
  send_servo( servo );
} 

void Client_servo_button::servoMiddle() {
  const char *servo = "middle";
  send_servo( servo );
}

void Client_servo_button::servoStop() {
  const char *servo = "stop";
  send_servo( servo );
}

int main(int argc, char *argv[]) {
    
  QApplication app(argc, argv);  
    
  Client_servo_button window;

  window.resize(300, 200);
  window.setWindowTitle("client servo button");
  window.show();

  return app.exec();
}
